docs/building/building.md

```diff
@@ -0,0 +1,136 @@
+        $ make clean-bpf-nocore     	# wipe ./dist/tracee.bpf.XXX.o
+        $ make clean-tracee-ebpf    	# wipe ./dist/tracee-ebpf
+        $ make clean-tracee-rules   	# wipe ./dist/tracee-rules
+        $ make clean-rules          	# wipe ./dist/rules
+    
+    # test
+    
+        $ make test-types           	# run unit tests for types module
+        $ make test-unit            	# run unit tests
+        $ make test-integration     	# run integration tests
+        $ make test-rules           	# opa test (tracee-rules)
+    
+    # flags
+    
+        $ STATIC=1 make ...                 # build static binaries
+        $ BTFHUB=1 STATIC=1 make ...        # build static binaries, embed BTF
+    ```
+
+4. Build **all** targets at once (but bpf-nocore)
+
+    ```text
+    $ make all
+    Submodule 'libbpf' (https://github.com/libbpf/libbpf.git) registered for path '3rdparty/libbpf'
+    Cloning into '/home/rafaeldtinoco/tracee/3rdparty/libbpf'...
+    mkdir -p dist/rules
+    GOOS=linux CC=clang GOARCH=amd64 CGO_CFLAGS= CGO_LDFLAGS= go build \
+        --buildmode=plugin \
+        -o dist/rules/builtin.so \
+        signatures/golang/export.go signatures/golang/kubernetes_api_connection.go signatures/golang/stdio_over_socket.go
+    ```
+
+5. Build a **static binary** by setting `STATIC=1`
+
+    ```text
+    $ STATIC=1 make all
+    CC="clang" \
+        CFLAGS=""-fPIC"" \
+        LD_FLAGS="" \
+        make \
+        -C ./3rdparty/libbpf/src \
+        BUILD_STATIC_ONLY=1 \
+        DESTDIR=/home/rafaeldtinoco/tracee/dist/libbpf \
+        OBJDIR=/home/rafaeldtinoco/tracee/dist/libbpf/obj \
+        INCLUDEDIR= LIBDIR= UAPIDIR= prefix= libdir= \
+        install install_uapi_headers
+    ...
+    ```
+
+6. Build a **static binary** with [BTFHUB Support](https://github.com/aquasecurity/btfhub)
+
+    ```text
+    $ BTFHUB=1 STATIC=1 make all
+    Cloning into '/home/rafaeldtinoco/tracee/3rdparty/btfhub'...
+    remote: Enumerating objects: 205, done.
+    remote: Counting objects: 100% (16/16), done.
+    remote: Compressing objects: 100% (12/12), done.
+    remote: Total 205 (delta 4), reused 10 (delta 3), pack-reused 189
+    Receiving objects: 100% (205/205), 10.59 MiB | 7.56 MiB/s, done.
+    Resolving deltas: 100% (73/73), done.
+    Cloning into '/home/rafaeldtinoco/tracee/3rdparty/btfhub-archive'...
+    remote: Enumerating objects: 1993, done.
+    remote: Counting objects: 100% (28/28), done.
+    remote: Compressing objects: 100% (23/23), done.
+    Receiving objects:  15% (301/1993), 154.97 MiB | 15.72 MiB/s
+    ```
+
+    !!! Note
+        BTFHUB support will embed several very small files (BTF files) into your
+        final binary. Those files will allow **tracee-ebpf** binary to be executed
+        in kernels that doesn't have embedded BTF information (the ones described
+        at the BTFHUB repository)
+
+    !!! Attention
+        compiling `tracee-rules` with STATIC=1 won't allow you to use golang based
```

Comment:
Isn't this right only for go plugins? What about go signatures that are compiled as part of tracee-rules?

Reply:
Yep, I changed it to:

```
!!! Attention
    compiling `tracee-rules` with STATIC=1 won't allow you to use golang based
    signatures as plugins, only as built-ins:
```